uefi/src/proto/debug/mod.rs

```diff
@@ -170,3 +170,46 @@ pub enum ProcessorArch: u32 => {
     /// RISC-V 128-bit
     RISCV_128   = 0x5128,
 }}
+
+/// The debug port protocol abstracts the underlying debug port
+/// hardware, whether it is a regular Serial port or something else.
+#[repr(C)]
+#[unsafe_protocol("eba4e8d2-3858-41ec-a281-2647ba9660d0")]
+pub struct DebugPort {
+    reset: extern "efiapi" fn(this: &DebugPort) -> Status,
+    write: extern "efiapi" fn(
+        this: &DebugPort,
+        timeout: u32,
+        buffer_size: &mut usize,
+        buffer: *const c_void,
+    ) -> Status,
+    read: extern "efiapi" fn(
+        this: &DebugPort,
+        timeout: u32,
+        buffer_size: &mut usize,
+        buffer: *mut c_void,
+    ) -> Status,
+    poll: extern "efiapi" fn(this: &DebugPort) -> Status,
+}
+
+impl DebugPort {
+    /// Flushes the port.
+    pub fn reset(&self) -> Result {
+        (self.reset)(self).into()
+    }
+
+    /// Writes 8 bytes at a time to the underlying SerialIo protocol.
+    pub fn write(&self, timeout: u32, buffer_size: &mut usize, buffer: *const c_void) -> Result {
+        (self.write)(self, timeout, buffer_size, buffer).into()
+    }
+
+    /// Reads data from the underlying SerialIo protocol.
+    pub fn read(&self, timeout: u32, buffer_size: &mut usize, buffer: *mut c_void) -> Result {
+        (self.read)(self, timeout, buffer_size, buffer).into()
+    }
+
+    /// Checks if the underlying port is read for data transfer.
```

Comment:
Change doc to "Check to see if any data is available to be read from the debugport device."

("Transfer" doesn't indicate the direction; I think poll only indicates a read can complete.)